exporter/exporterhelper/internal/base_exporter.go

```diff
@@ -93,6 +93,25 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, osf ObsrepSe
 		return nil, err
 	}
 
+	if be.BatcherCfg.Enabled {
+		bs := NewBatchSender(be.BatcherCfg, be.Set, be.BatchMergeFunc, be.BatchMergeSplitfunc)
+		for _, opt := range be.BatcherOpts {
+			err = multierr.Append(err, opt(bs))
+		}
+
+		if bs.mergeFunc == nil || bs.mergeSplitFunc == nil {
+			err = multierr.Append(err, fmt.Errorf("WithRequestBatchFuncs must be provided for the batcher applied to the request-based exporters"))
+		}
+
+		// Setting these callbacks to base exporter, because if they come in from options, these are only set to batch sender.
+		be.BatchMergeFunc = bs.mergeFunc
+		be.BatchMergeSplitfunc = bs.mergeSplitFunc
+
+		if !be.queueCfg.Enabled {
+			be.BatchSender = bs
```

Comment:
Do you need to create `bs` at all if not used for `!be.queueCfg.Enabled`?

Reply:
The only reason this exist is to set up `batchMergeFunc`. `batchMergeFunc` comes in as a batcher option 

https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/internal/base_exporter.go#L33

which has to be set on a batchSender. Maybe we should deprecate this interface now that a `batchsender` is not necessarily created. Let me work on a separate PR for this